### ISSUE TITLE
ungrouped-imports false positive in `try:` block

### ISSUE THREAD

**Issue by asottile** [PyCQA/pylint#2225]
### Steps to reproduce

I'm intentionally separating the noisy `try` / `except` from the rest of the imports and I don't believe `pylint` should be flagging this as `ungrouped-imports`

```python
from yaml import safe_load

import mypkg

try:
    from yaml.cyaml import CParser as Parser
except ImportError:
    from yaml.parser import Parser
```

(note that this style is chosen so import reorderers have an easier time managing where imports should go and to keep the "sometimes" code away from the "always" code)

### Current behavior

```
$ pylint --disable=unused-import,import-error,missing-docstring,invalid-name --rcfile=/dev/null test.py  
Using config file /dev/null
************* Module test
C:  6, 4: Imports from package yaml are not grouped (ungrouped-imports)

------------------------------------------------------------------
Your code has been rated at 8.33/10 (previous run: 8.33/10, +0.00)
```

### Expected behavior

(no error)

### pylint --version output

```
$ pylint --version
Using config file /private/tmp/cfn-python-lint/pylintrc
pylint 1.9.2, 
astroid 1.6.5
Python 3.6.5 (default, Mar 30 2018, 06:41:53) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

(originally from https://github.com/awslabs/cfn-python-lint/pull/171)

**Comment by PCManticore** [comment]
Hi, thanks for the report! I'm not entirely sure why is this a bug, wouldn't moving the last `yaml` import  group next to the first `yaml` make more sense, as they both represent third party imports?

**Comment by asottile** [comment]
it would, however that throws off import reordering tools and intermixes ugly code with clean code which I think is worse :)

**Comment by rik** [comment]
I'm running into the same issue with `if TYPE_CHECKING:` imports.

**Comment by ewerybody** [comment]
I'd underline this one! Have a similar issue tho. I WANT the **`ungrouped-imports`** checks but:

* not in `try-except` blocks and:
* not in `if __name__ == '__main__':`-blocks

because when I run a module as script to test itself and I'd like to import something for this .. well it's kinda frustrating to also provide a `# pylint: disable=` right away all the time.
